diff --git a/superset-frontend/src/features/home/DashboardTable.tsx b/superset-frontend/src/features/home/DashboardTable.tsx
index 01b385138d..d096aa5d39 100644
--- a/superset-frontend/src/features/home/DashboardTable.tsx
+++ b/superset-frontend/src/features/home/DashboardTable.tsx
@@ -159,14 +159,6 @@ function DashboardTable({
         setItem(LocalStorageKeys.HomepageDashboardFilter, TableTab.Favorite);
       },
     },
-    {
-      name: TableTab.Mine,
-      label: t('Mine'),
-      onClick: () => {
-        setActiveTab(TableTab.Mine);
-        setItem(LocalStorageKeys.HomepageDashboardFilter, TableTab.Mine);
-      },
-    },
   ];
 
   if (otherTabData) {
