diff --git a/superset/utils/webdriver.py b/superset/utils/webdriver.py
index f9a7899ff9..10688b4648 100644
--- a/superset/utils/webdriver.py
+++ b/superset/utils/webdriver.py
@@ -191,7 +191,6 @@ class WebDriverPlaywright(WebDriverProxy):
                     # chart containers didn't render
                     logger.debug("Wait for chart containers to draw at url: %s", url)
                     slice_container_locator = page.locator(".chart-container")
-                    slice_container_locator.first.wait_for()
                     for slice_container_elem in slice_container_locator.all():
                         slice_container_elem.wait_for()
                 except PlaywrightTimeout:
