diff --git a/plugins/poll/app/models/poll.rb b/plugins/poll/app/models/poll.rb
index e4660307915..410a073bd61 100644
--- a/plugins/poll/app/models/poll.rb
+++ b/plugins/poll/app/models/poll.rb
@@ -9,11 +9,11 @@ class Poll < ActiveRecord::Base
   has_many :poll_options, -> { order(:id) }, dependent: :destroy
   has_many :poll_votes
 
-  enum :type, { regular: 0, multiple: 1, number: 2, ranked_choice: 3 }, scopes: false, instance_methods: false
-  enum :status, { open: 0, closed: 1 }, scopes: false, instance_methods: false
-  enum :results, { always: 0, on_vote: 1, on_close: 2, staff_only: 3 }, scopes: false, instance_methods: false
-  enum :visibility, { secret: 0, everyone: 1 }, scopes: false, instance_methods: false
-  enum :chart_type, { bar: 0, pie: 1 }, scopes: false, instance_methods: false
+  enum :type, { regular: 0, multiple: 1, number: 2, ranked_choice: 3 }, scopes: false
+  enum :status, { open: 0, closed: 1 }, scopes: false
+  enum :results, { always: 0, on_vote: 1, on_close: 2, staff_only: 3 }, scopes: false
+  enum :visibility, { secret: 0, everyone: 1 }, scopes: false
+  enum :chart_type, { bar: 0, pie: 1 }, scopes: false
 
   validates :min, numericality: { allow_nil: true, only_integer: true, greater_than_or_equal_to: 0 }
   validates :max, numericality: { allow_nil: true, only_integer: true, greater_than: 0 }
